invoice/src/main/java/org/killbill/billing/invoice/InvoiceDispatcher.java

```diff
@@ -571,7 +571,30 @@ public boolean apply(final InvoiceItem input) {
                     if (exitingItem != null) {
                         invoice.removeInvoiceItem(exitingItem);
                     }
-                    invoice.addInvoiceItem(cur);
+
+                    final InvoiceItem sanitizedInvoiceItemFromPlugin = new InvoiceItemCatalogBase(cur.getId(),
+                                                                                                  cur.getCreatedDate(),
+                                                                                                  MoreObjects.firstNonNull(cur.getInvoiceId(), invoice.getId()),
```

Comment:
Interesting, so does this mean that we allow to add additional items outside or the current invoice being generated?

Reply:
Yes, but this is not new (I believe we allowed this when we implemented the `UPDATE` feature, see https://github.com/killbill/killbill/issues/807).